packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1367,15 +1358,148 @@ class RouteObserver<R extends Route<dynamic>> extends NavigatorObserver {
 abstract class RouteAware {
   /// Called when the top route has been popped off, and the current route
   /// shows up.
-  void didPopNext() { }
+  void didPopNext() {}
 
   /// Called when the current route has been pushed.
-  void didPush() { }
+  void didPush() {}
 
   /// Called when the current route has been popped off.
-  void didPop() { }
+  void didPop() {}
 
   /// Called when a new route has been pushed, and the current route is no
   /// longer visible.
-  void didPushNext() { }
+  void didPushNext() {}
 }
+
+class _DialogRoute<T> extends PopupRoute<T> {
+  _DialogRoute({
+    @required RoutePageBuilder pageBuilder,
+    bool barrierDismissible = true,
+    String barrierLabel = 'Dismiss',
```

Comment:
This string needs localization.

Reply:
Looking at this a little more: It looks like _DialogRoute is only every instantiated in showGeneralDialog, which will always pass in a value for barrierLabel. So the default value here should never be in use. Can you just remove it?

Reply:
This is true - related, would it be a good idea to make barrierLabel a required argument in showGeneralDialog, or would that be unnecessary?

Reply:
I'd just not default and assert that if it's barrierDismissible, the user needs to pass in a non-null barrierLabel as well.

Reply:
+1 to @xster